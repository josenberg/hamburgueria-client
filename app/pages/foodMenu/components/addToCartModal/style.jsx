export const modalStyles = {
  overlay: {},
  content: {
    width: '400px',
    height: '390px',
    margin: '0 auto',
  },
};

export const title = {
  textAlign: 'center',
  margin: '10px 0 0 0',
};

export const titleContainer = {
  border: '5px solid #7F5417',
  backgroundColor: '#7F171F',
  color: 'white',
  fontFamily: "'Lalezar', cursive",
  width: '390px',
  margin: '0 auto 30px auto',
};

export const controlsContainer = {
  margin: '30px 0 0 0',
};
