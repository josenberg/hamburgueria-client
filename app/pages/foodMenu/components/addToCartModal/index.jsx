import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { isNil } from 'ramda';
import ReactModal from 'react-modal';

import IngredientsPicker from '@/components/ingredientsPicker';
import Grid from '@/components/grid';

import { addToCart } from '@/state/cart/actions';
import { updateItemIngredientQuantity } from '@//utilities/order';

import * as styles from './style';

const AddToCartModal = ({
  item,
  closeModal,
  updateItem,
  ingredients,
  rules,
  addItemToCart,
}) => (
  <ReactModal
    isOpen={!isNil(item)}
    style={styles.modalStyles}
    ariaHideApp={false}
  >
    {!isNil(item) ? (
      <Grid rows="auto auto auto" columns="1fr">
        <Grid.Cell row="1" column="1" style={styles.titleContainer}>
          <h1 style={styles.title}> Adicionar ao carrinho </h1>
        </Grid.Cell>
        <Grid.Cell row="2" column="1">
          <IngredientsPicker
            ingredients={ingredients}
            item={item}
            rules={rules}
            updateItem={(ingredientId, quantity) => {
              updateItemIngredientQuantity(updateItem, ingredients, item, ingredientId, quantity);
            }}
            onRefuse={closeModal}
            onAccept={() => {
              addItemToCart(item);
              closeModal();
            }}
          />
        </Grid.Cell>
      </Grid>
    ) : null}
  </ReactModal>
);

AddToCartModal.propTypes = {
  item: PropTypes.object,
  ingredients: PropTypes.array,
  rules: PropTypes.array,
  closeModal: PropTypes.func.isRequired,
  updateItem: PropTypes.func.isRequired,
  addItemToCart: PropTypes.func.isRequired,
};

const mapDispatchToProps = {
  addItemToCart: addToCart,
};

export default connect(null, mapDispatchToProps)(AddToCartModal);
