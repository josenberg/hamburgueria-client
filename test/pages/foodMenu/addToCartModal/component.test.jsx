import 'babel-polyfill';
import React from 'react';
import renderer from 'react-test-renderer';

import addToCartModal from '../../../../app/pages/foodMenu/components/addToCartModal';

test('Add To Cart Modal should render without issues', () => {
  const component = renderer.create(
    <addToCartModal
      ingredients={[]}
      rules={[]}
      item={{}}
      isOpen={false}
    />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
