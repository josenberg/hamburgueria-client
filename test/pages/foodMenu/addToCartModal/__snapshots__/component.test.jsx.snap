// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add To Cart Modal should render without issues 1`] = `
<addToCartModal
  ingredients={Array []}
  isOpen={false}
  item={Object {}}
  rules={Array []}
/>
`;
